tsfc/driver.py

```diff
@@ -198,6 +198,8 @@ def compile_integral(integral_data, form_data, prefix, parameters,
     expressions = impero_utils.preprocess_gem(expressions, **options)
     assignments = list(zip(return_variables, expressions))
 
+    builder.register_tabulations(expressions)
```

Comment:
This function needs to be exposed (even just stubbed out) in all the kernel interface files, so that tsfc still works for dolfin.

Reply:
I see an emerging pattern here. Sometimes the kernel interface needs to investigate the optimised GEM expression to see if certain externally provided data is required. This is the case for:

* Cell orientations
* Cell sizes (#167)
* Externally provided tabulations (here; for Themis)

Perhaps it may make sense to chuck them behind a single method of the kernel interface, since they don't clutter the driver code, but remain in the kernel interface, which is what they are relevant to.

Reply:
Good idea, I will have a go here.